tendrl/ceph_integration/ceph.py

```diff
@@ -392,49 +396,97 @@ def ceph_command(cluster_name, command_args):
-        args = ["rbd", "--pool", pool_name, "--cluster", cluster_name] + \
-            command_args
+def rbd_operation(cluster_name, operation, attr, pool_name, result, queue):
+    import rados
+    cluster = rados.Rados(name=RADOS_NAME,
+                          clustername=cluster_name,
+                          conffile='')
+    cluster.connect()
+    
+    try:
+        ioctx = cluster.open_ioctx(pool_name)
+        try:
+            if operation == "create":
+                rbd_inst = rbd.RBD()
+                rbd_inst.create(ioctx, attr["name"], int(attr["size"]))
+            if operation == "delete":
+                rbd_inst = rbd.RBD()
+                rbd_inst.remove(ioctx, attr["name"])
+            if operation == "update":
+                try:
+                    image = rbd.Image(ioctx, attr["name"])
+                    size  = int(attr['size'])
+                    image.resize(size)
+                finally:
+                    image.close()
+            if operation == "list":
+                rbd_inst = rbd.RBD()
+                result['out'] = rbd_inst.list(ioctx)
+            if operation == "info":
+                try:
+                    result['out'] = {}
+                    image = rbd.Image(ioctx, attr['name'])
+                    size = int(image.stat()['size'])
+                    result['out']['size'] = size
+                    flags = int(image.flags())
+                    result['out']['flags'] = flags
+                finally:
+                    image.close()
+            if operation == "usage":
+                args = "du --image %s" % attr['name']
+                cmd_args = "rbd --pool %s --cluster %s " % (pool_name, cluster_name) + \
+                    args
+                cmd  = cmd_utils.Command(cmd_args)
+                out, err, rc = cmd.run()
+                if not err:
+                    result['out'] = out
+                elif out:
+                    # When fast-diff map is not enabled
+                    # It gives output with error
+                    result['out'] = out
+                else:
+                    result['err'] = err
+                    result['status'] = 1
+        except (TypeError,
+                rbd.InvalidArgument,
+                rbd.ImageExists,
+                rbd.ImageNotFound,
+                rbd.ImageBusy,
+                rbd.ImageHasSnapshots,
+                rbd.FunctionNotSupported,
+                rbd.ArgumentOutOfRange) as ex:
+            result['err'] =  str(ex)
+            result['status'] = 1
+        finally:
+            ioctx.close()
+            
+    finally:
+        cluster.shutdown()
+    queue.put(result)
+
+def rbd_command(cluster_name, operation, attr, pool_name):
+    # Starting as a process
+    result = {}
+    result['status'] = 0
+    result['err'] = ''
+    result['out'] = ''
+    # Exchanging objects between processes
+    queue = multiprocessing.Queue()
+    p = multiprocessing.Process(
```

Comment:
i will change that to gevent, multiprocessing timout is works like it wait unit process finish if the process finished before the specified time it automatically stops wait and executes further. that why i used this.
any way i will changed to gevent